portfolio/src/main/webapp/script.js

```diff
@@ -16,3 +16,18 @@ function openModal(num){
     $('#modalMyDog').modal('show'); 
     document.getElementById("modalMyDogImg").src = 'images/cuky'+num+'.jpg';
 }
+
+let fetchFunction = () => {
```

Comment:
Use const instead of let here since you're not reassigning fetchFunction later on.
Actually, why are you using arrow function instead of: function fetchData() {...}, similar to your createListElement function below?